common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -124,11 +125,14 @@ protected[core] object Exec
 
             val obj = v.asJsObject
 
-            val kind = obj.getFields("kind") match {
+            val kindField = obj.getFields("kind") match {
                 case Seq(JsString(k)) => k.trim.toLowerCase
                 case _                => throw new DeserializationException("'kind' must be a string defined in 'exec'")
             }
 
+            // map "default" virtual runtime versions to the currently blessed actual runtime version
+            val kind = resolveDefaultRuntime(kindField)
```

Comment:
could rewrite to `val kind = resolveDefaultRuntime(obj.getFields("kind") ... from above)`